File: src/main/ruby/OnBoarding.rb
STDOUT.sync = true # DO NOT REMOVE
# CodinGame planet is being attacked by slimy insectoid aliens.
# <---
# Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.


# game loop
loop do
  enemy_1 = gets.chomp # name of enemy 1
  dist_1 = gets.to_i # distance to enemy 1
  enemy_2 = gets.chomp # name of enemy 2
  dist_2 = gets.to_i # distance to enemy 2

  if dist_1 >= dist_2 then
    puts enemy_2
  elsif dist_2 > dist_1 then
    puts enemy_1
  else
    puts enemy_1
  end
end

File: src/main/java/OnBoarding.java
import java.util.Scanner;

/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 *
 * @author psenger
 * @version 1.0
 * @see
 * @since 11/14/15
 */
public class OnBoarding {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        // game loop
        while (true) {
            String enemy1 = in.next(); // name of enemy 1
            int dist1 = in.nextInt(); // distance to enemy 1
            String enemy2 = in.next(); // name of enemy 2
            int dist2 = in.nextInt(); // distance to enemy 2

            // Write an action using System.out.println()
            if ( dist1 >= dist2 ) {
                System.out.println(enemy2);
            } else if (dist2 > dist1 ){
                System.out.println(enemy1);
            } else {
                System.out.println(enemy1);
            }
        }
    }
}
